### PR TITLE
Add the option to build without exceptions

### PR DESCRIPTION
**Pull request by godlikepanos**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -56,17 +62,23 @@ target_include_directories(spirv-cross-core PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 if (NOT "${MSVC}")
   set(spirv-compiler-options -std=c++11 -Wall -Wextra -Werror -Wshadow)
   set(spirv-compiler-defines __STDC_LIMIT_MACROS)
+
+  set(spirv-exe-options ${spirv-compiler-options})
```
Comment (HansKristian-ARM) [inline CMakeLists.txt]: I think it makes sense to propagate -fno-exceptions even to executable to be able to test this more easily.

File: CMakeLists.txt [hunk 2]
```diff
@@ -16,6 +16,12 @@ cmake_minimum_required(VERSION 2.8)
 project(SPIRV-Cross)
 enable_testing()
 
+option(SPIRV_CROSS_EXCEPTIONS_TO_ASSERTIONS "Instead of throwing exceptions assert. Only applies to the library targets" OFF)
+
+if(SPIRV_CROSS_EXCEPTIONS_TO_ASSERTIONS)
+	add_definitions(-DSPIRV_CROSS_EXCEPTIONS_TO_ASSERTIONS)
```
Comment (HansKristian-ARM) [inline CMakeLists.txt]: Makes sense to document this in README as well.

File: Makefile [hunk 3]
```diff
@@ -15,7 +15,7 @@ CXXFLAGS += -std=c++11 -Wall -Wextra -Wshadow -D__STDC_LIMIT_MACROS
 ifeq ($(DEBUG), 1)
 	CXXFLAGS += -O0 -g
 else
-	CXXFLAGS += -O2 -g
+	CXXFLAGS += -O2 -DNDEBUG
```
Comment (HansKristian-ARM) [inline Makefile]: A trivial ifeq ($(EXCEPTIONS_TO_ASSERTIONS), 1) CXXFLAGS += -D ... -fno-exceptions endif could work fine too.

File: CMakeLists.txt (hunk 1 above)
Comment (godlikepanos) [inline CMakeLists.txt]: Considered that as well. That will require some refactoring of main.cpp to remove the exceptions from there as well. Happy to do it.

Comment (HansKristian-ARM) [inline CMakeLists.txt]: Ye, makes sense.

File: spirv_common.hpp [hunk 4]
```diff
@@ -17,13 +17,27 @@
...
 #include <functional>
 #include <sstream>
-#include <stdio.h>
-#include <string.h>
 
 namespace spirv_cross
 {
+
+#ifdef SPIRV_CROSS_EXCEPTIONS_TO_ASSERTIONS
+[[noreturn]] inline void report_and_abort(const std::string &msg)
```
Comment (HansKristian-ARM) [inline spirv_common.hpp]: [[noreturn]] breaks MSVC it seems, so an #ifdef is probably in order here.

**Pull request merged**